Telegram/SourceFiles/ui/text/text_entity.cpp

```diff
@@ -2120,9 +2135,11 @@ void PrepareForSending(TextWithEntities &result, qint32 flags) {
 		ParseEntities(result, flags);
 	}
 
-	ReplaceStringWithChar(qstr("--"), QChar(8212), result, true);
-	ReplaceStringWithChar(qstr("<<"), QChar(171), result);
-	ReplaceStringWithChar(qstr(">>"), QChar(187), result);
+	if (cMessageFormatting()) {
+		ReplaceStringWithChar(qstr("--"), QChar(HtmlAlias_mdash), result, true);
+		ReplaceStringWithChar(qstr("<<"), QChar(HtmlAlias_laquo), result);
+		ReplaceStringWithChar(qstr(">>"), QChar(HtmlAlias_raquo), result);
```

Comment:
Honestly I can't see a point behind using `HtmlAlias_mdash` instead of just `EmDash`.

Reply:
I suggest to remove `Html` stuff if we don't have any better reasons to keep it. After all, how's that related to HTML?